### DISCUSSION TITLE
Welcome to koseven Discussions!

### DISCUSSION THREAD

**Discussion by neo22s** [koseven/koseven#439] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rahul-works** [comment]
thanks, My name is Rahul Thakur. I have 12 + years of work experience, I have a PHP/NodeJs Developer. After covid, I got hooked to GitHub and I love to contribute to the GitHub Project. I was a Kohana user. So, I use koSeven as well. 

Rahul
Bangalore, India
https://www.linkedin.com/in/rahulthakur14/

**Reply by neo22s** [reply]
welcome!!! ;)

**Comment by GrantFordAUS** [comment]
Hello koseven community,
My name is Grant Ford I live in Tasmania, Australia and own a site called https://www.scanlink.com.au/
The website was built several years ago using an old version of kohana and I'd like to update the site to eliminate many of the problems currently occurring.
My expertise is in sales & marketing so I don't have any tech/IT skills. I originally engaged the services of a company in India to build the site for me. The same company has suggested that I should now re-build the site using Laravel with Vue.js 
I am wondering if updating the site to koseven would be quicker and more economical?
If anyone could help or offer any suggestions I would sincerely appreciate it.
many thanks :)
Grant
grant.ford@galatech.com.au

**Reply by svenbw** [reply]
Grant,
It depends. From my experience some migrations are very simple, other are harder.

Before it was common practice that Kohana modules included a vendor directory containing the code from another repo (for pdf generation, email, ...). The vendor code can be old and might even break if you run it on a more recent version of PHP.
Now we support composer, which makes (future) updates very simple.

A second factor is the amount of business logic that is embedded in the application. And how many issues occur if you switch to PHP 7.4. Without looking in the codebase it's very difficult to give you any advise.

If you compare the technologies: Laravel/Vue.js will probably be an SPA implementation, which gives a "smoother" user experience. But server side rendered applications have their own place and still fullfil the requirements of an application and are therefore certainly not a bad choice.

If you find an experienced PHP developer which knows the basics of routing/views/controllers Koseven is still a valid choice and there is a very supportive (Telegram) community that is very supportive.

**Reply by neo22s** [reply]
Totally agree with @svenbw 

I think migrating to Koseven will be cheaper in the short term for sure...

**Reply by GrantFordAUS** [reply]
Gentlemen,
I sincerely appreciate you taking the time to reply.
I would like to investigate migrating to Koseven - Could either of you recommend a good developer that may be able to perform the migration and provide ongoing support?

Cheers
Grant

**Reply by jstrobel** [reply]
Hi Grant,
thanks for reaching out the koseven group.
I totally agree my previous speaker friends.

I think a good place to ask for a developer who can help you is the telegram group - right now there are around 100 members - more or less active :)
But a lot of people with a good experience in koseven.

greetings
jens

**Comment by sttt** [comment]
Made a fix to the file. In init 'errors' => FALSE, and the error was generated:
\htdocs\system\classes\Kohana\Request\Client\Internal.php
line 114
```
if (Kohana::$errors === TRUE) 
{
	// Generate an appropriate Response object
	$response = Kohana_Exception::_handler($e);
}
```

**Comment by nicolas-grzybowski** [comment]
Hi !
I'm Nicolas, French Developer since 15 years ago. I use Koseven since a long times. I have 2 big projects run on it and if necessary I can help the community to keep healthy
Have a good day !

**Reply by shiroorg** [reply]
Then could you suggest some ideas for improvement?

**Reply by neo22s** [reply]
Support for php 8 and php 8.1 should be simple to achieve to the last stable release.

**Comment by timhj** [comment]
Yes but then the real question... koeight?
